include/CL/sycl/platform/detail/platform_tail.hpp

```diff
@@ -39,7 +39,10 @@ platform::get_devices(info::device_type device_type) const {
 
```

Comment:
The issue above (I cannot comment it in GitHub :-( ) is that you used `boost::compute::system::devices()` instead of `get_boost_compute()->devices()` (see `CL/device.hpp` to steal the implementation of `get_boost_compute()`).

Furthermore it does not make sense to have the default device here:
```C++
// Start with the default device
vector_class<device> devices = { {} };
```

Just keep `vector_class<device> devices;`